common/common.go

```diff
@@ -270,8 +274,11 @@ func ExtractHost(address string) string {
 
 // ExtractPort returns the port name out of a string
 func ExtractPort(host string) int {
-	portStr := strings.Split(host, ":")[1]
-	port, _ := strconv.Atoi(portStr)
+	portStrs := strings.Split(host, ":")
+	if len(portStrs) == 1 {
```

Comment:
```
func TestSplit(t *testing.T) {
	hi := strings.Split("", ":")
	t.Log(len(hi))
}
```
There is no chance, but I will change the return to default to port 80 instead of -1